src/irc.cc

```diff
@@ -69,6 +70,22 @@ void addmessage(struct ctx *ctxptr, char *nick, const char *text) {
 		}
 		lastmsg->next = newmsg;
 	}
+	if(ctxptr->max_msg_lines>0) {
+		int msgcnt = 0;
+		newmsg = ctxptr->messages;
+		while(newmsg) {
```

Comment:
I'm not sure, what variable should be checked. The loop is pretty similar to the one in line 68..70. Or do you mean, I should check for an overflow of `msgcnt`? This, however, only increases by one each time, the function is called, and gets truncated to `ctxptr->max_msg_lines` at the end of the function call iff `ctxptr->max_msg_lines>0`. Should I prepare the function for a changing `ctxptr->max_msg_lines`, so it may switch from 0 to some positive value while the current message count is larger than max(int)?